tests/Unit/Domain/Creators/Test_Sphere.cpp

```diff
@@ -784,42 +770,130 @@ void test_kerr_horizon_conforming(const bool use_time_dependence) {
+    get<0>(x) = local_radius * sin(get<0>(local_theta_phi)) *
+                cos(get<1>(local_theta_phi));
+    get<1>(x) = local_radius * sin(get<0>(local_theta_phi)) *
+                sin(get<1>(local_theta_phi));
+    get<2>(x) = local_radius * cos(get<0>(local_theta_phi));
+    return x;
+  };
+
+  const DataVector radius =
+      get(gr::Solutions::kerr_schild_radius_from_boyer_lindquist(
+          inner_radius, theta_phi, mass, spin));
+
+  {
+    INFO(
+        "Check that inner radius is deformed to constant Boyer-Lindquist "
+        "radius");
+
+    // Set up coordinates on an ellipsoid of constant Boyer-Lindquist radius
+    const auto x = set_x(radius, theta_phi);
+
+    // Time dependence
+    time_dependent_options = std::make_unique<
+        domain::creators::time_dependence::Shape<domain::ObjectLabel::None>>(
+        time, l_max, mass, spin, std::array<double, 3>{{0., 0., 0.}},
+        inner_radius, 4.);
+
+    test_shape_distortion_general(time, std::move(time_dependent_options),
+                                  inner_radius, x);
+
+    // KerrSchild-BoyerLindquist. Use same x as above
+    time_dependent_options = domain::creators::sphere::TimeDependentMapOptions{
+        time,
+        {{l_max, domain::creators::time_dependent_options::
+                     KerrSchildFromBoyerLindquist{mass, spin}}},
+        std::nullopt,
+        std::nullopt,
+        std::nullopt};
+
+    test_shape_distortion_general(time, std::move(time_dependent_options),
+                                  inner_radius, x);
+  }
+
+  {
+    INFO("Check reading in Ylms from file");
+    const std::string h5_filename{"StrahlkorperCoefsFile.h5"};
+    const std::string subfile_name{"Ylm_coefs"};
+    if (file_system::check_if_file_exists(h5_filename)) {
+      file_system::rm(h5_filename, true);
+    }
+    ylm::Strahlkorper<Frame::Distorted> strahlkorper{l_max, l_max, radius,
+                                                     std::array{0.0, 0.0, 0.0}};
+    std::vector<std::string> legend{};
+    std::vector<double> data{};
+    ylm::fill_ylm_legend_and_data(make_not_null(&legend), make_not_null(&data),
+                                  strahlkorper, time, l_max);
+    {
+      h5::H5File<h5::AccessType::ReadWrite> test_file{h5_filename, true};
+      auto& subfile = test_file.insert<h5::Dat>("/" + subfile_name, legend);
+      subfile.append(data);
+      test_file.close_current_object();
+    }
+
+    time_dependent_options = domain::creators::sphere::TimeDependentMapOptions{
+        time,
+        {{l_max,
+          domain::creators::time_dependent_options::YlmsFromFile{
+              h5_filename, std::vector{subfile_name}, time, std::nullopt,
+              false},
+          // Constructing a strahlkorper from collocation radii will not exactly
+          // match the collocation points (see Strahlkorper constructor docs).
+          // For this reason, the 00 coef we calculate is not exact. This is why
```

Comment:
Does this mean the round trip (write shape coeffs to file, then read them back in) is not exact? If so please note that caveat down in docs and option help, and we should probably fix that.

Reply:
No it's not the writing/reading, it's just constructing the strahlkorper from collocation radii that apparently isn't exact. It's noted in the Strahlkorper constructor and I noted it here

Reply:
Oh and in the round trip we write out the Strahlkorper coefficients directly. Sounds good then.